### PR TITLE
Filter service

### LINKED ISSUE #97: Persist filter options in local storage
Implement a FilterService that sets the filter when the app is starting up and persists the settings in the local storage.

Would be a good contribution for someone who is looking to contribute @PokemonGoers/catch-em-all

### LINKED ISSUE #101: Map filtering
Since the filter is now well-synced throughout the app, we need to pass it to the pokemap.
This requires transforming the Filter() to a filterObj{} according to the specification (https://github.com/PokemonGoers/Catch-em-all/issues/24).

> - `filter({ pokemonIds, sightingsSince, predictionsUntil })` applies filter options to the map.
>   - `pokemonIds` Array of Pokemon IDs which are shown (applies to predictions and sightings) on the map. If `null` show all 151 Pokemon.
>   - `sightingsSince` Time in seconds for which sightings (includes mobs) in the past are shown. If `0` don't show any sightings.
>   - `predictionsUntil` Time in seconds for which predictions are shown. If `0` don't show any predictions.

Predictions and Sightings should be passed as Date().
Ideally, the lookup / conversion should be done in a Service.

@PokemonGoers/catch-em-all

### PR DESCRIPTION
**Pull request by johartl**
- Translate between internal representation of `sightingsRange`/`predictionsRange` and the corresponding filter options `sightingsSince`/`predictionsUntil`
- Persists filter options in native storage
- A bit of refactoring

Closes #101 and closes #97

### CODE REVIEW

**Review by AlexanderLill — APPROVED** [review]

**Comment by sacdallago** [comment]
@mfkaptan @MajorBreakfast @Lugitan @Georrgi you realize that you can review PRs as well, right? Since @AlexanderLill @WoH and @johartl seem to be putting a lot of effort into coding, having someone else actively reviewing, testing and correcting eventual bugs would not be so bad.

**Comment by gyachdav** [comment]
All, please remember that we are monitoring your contribution on GH. Slacking will translate into lower score in your final grade. No way around it.

**Review by mfkaptan — APPROVED** [review]

File: ionic2/config.xml [hunk 1]
```diff
@@ -1,61 +1,62 @@
-<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (WoH) [inline ionic2/config.xml:1]: This is a file that was carried over by RC0 I think. I don't think the current version uses this file.

Comment (johartl) [reply ionic2/config.xml:1]: No it has always been used to store installed cordova plugins and options (like the icons or the splash screen).

File: ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts [hunk 2]
```diff
@@ -31,16 +32,18 @@ export class PokeFilterPokemonTabComponent implements OnInit {
     pokemonTypes: []
   };
 
-  constructor(private apiservice: ApiService) {
+  constructor(private apiservice: ApiService, private filterService: FilterService) {
```
Comment (WoH) [inline ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts:35]: apiService, randomly noticed this just now, just reminding myself to rename this

**Comment by WoH** [comment]
Moving the lookup and static data into a service is very much needed, thank you. There are quite a few things going on though, maybe split it up next time if there is a chance. Extending the Redux/Event based style is very good as we talked about, wrapping the filter in a service that publishes the events doesn't only decrease the complexity by a lot it also looks pretty clean. 10/10

File: ionic2/config.xml (hunk 1 above)
Comment (WoH) [reply ionic2/config.xml:1]: Ok, my bad, I only ever noticed it on the RC0 branch.

File: ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts (hunk 2 above)
Comment (johartl) [reply ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts:35]: Rename to...?

Comment (WoH) [reply ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts:35]: apiservice -> apiService

Comment (johartl) [reply ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts:35]: Oh I never noticed

**Pull request merged**